java/foundation/src/org/xtreemfs/foundation/TimeSync.java

```diff
@@ -260,6 +267,13 @@ public static TimeSync initializeGPSD(InetSocketAddress gpsd, int timeSyncInterv
 
         TimeSync s = new TimeSync(ExtSyncSource.GPSD, null, gpsd, timeSyncInterval, localTimeRenew);
         s.start();
+
+        try {
+            s.waitForStartup();
+        } catch (Exception e) {
```

Comment:
I think catching this Exception is counterproductive. The system may misbehave if starting the TimeSync fails and should better crash immediately. Btw, please add a reference to the current object instead of "null" while logging errors.